server/src/main/java/org/opensearch/search/backpressure/trackers/CpuUsageTracker.java

```diff
@@ -60,7 +77,7 @@ public String name() {
     @Override
     public Optional<TaskCancellation.Reason> checkAndMaybeGetCancellationReason(Task task) {
         long usage = task.getTotalResourceStats().getCpuTimeInNanos();
-        long threshold = getCpuTimeNanosThreshold();
+        long threshold = (task instanceof SearchTask) ? getCpuTimeNanosThresholdForSearchQuery() : getCpuTimeNanosThreshold();
```

Comment:
What about `SearchShardTask`?

Reply:
We are assigning value to `threshold` variable based on the class type of the task (SearchTask or ShardShardTask). If it is `SearchTask`, it will be equal to `getCpuTimeNanosThresholdForSearchQuery()`, else it will be equal to `getCpuTimeNanosThreshold()`.

Reply:
We should move it to `SearchBackpressureTask` and let's each task type makes a decision for the right value

Reply:
Done.